### PR TITLE
Add distinct_1/2 metric

### PR DESCRIPTION
**Pull request by moshesbeta**
I add a new evaluation metric, Distinct 1/2, for the generate task evaluation. I have uploaded the new scripts "_answer_distinct12.py" and "test_answer_distinct12.py", and the modified version of "__init__.py".

### CODE REVIEW

File: rageval/metrics/_answer_distinct12.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
+from collections import Counter
+from dataclasses import dataclass
+from typing import List
+import datasets
+from nltk import ngrams
+from rageval.metrics import Metric, add_attribute
+
```
Comment (QianHaosheng) [inline rageval/metrics/_answer_distinct12.py:7]: `import/from` is not standardized here

File: rageval/metrics/_answer_distinct12.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        return datasets.MetricInfo(
+            description=_DESCRIPTION,
+            inputs_description=_KWARGS_DESCRIPTION,
+            citation=_CITATION,
+            features=datasets.Features(
+                {
+                    "hypothesis": datasets.Value("string"),
+                }
+            ),
+            codebase_urls=["https://github.com/Hannibal046/SelfMemory/blob/58d8b611ad51605091c7555c0f32dce6702dadbf/src/utils/metrics_utils.py"],
```
Comment (QianHaosheng) [inline rageval/metrics/_answer_distinct12.py:99]: reference link is better to be replaced by `https://github.com/Hannibal046/SelfMemory/blob/main/src/utils/metrics_utils.py`

File: rageval/metrics/_answer_distinct12.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        bigram_counter.update(ngrams(tokens, 2))
+
+    distinct_1 = len(unigram_counter) / sum(unigram_counter.values())
+    distinct_2 = len(bigram_counter) / sum(bigram_counter.values())
+
+    return {"distinct_1": distinct_1, "distinct_2": distinct_2}
+
+
+@dataclass
+@add_attribute('mtype', 'Diversity')
```
Comment (QianHaosheng) [inline rageval/metrics/_answer_distinct12.py:68]: maybe `Diversity` is not a good metric type

File: rageval/metrics/_answer_precision.py
Comment (QianHaosheng) [inline rageval/metrics/_answer_precision.py:1]: This metric is lacked of a unit test file.

`_answer_precision` is not a suitable file name. It is recommended to change it to `_answer_perplexity`

File: rageval/metrics/_answer_precision.py [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+from transformers import AutoModelForCausalLM, AutoTokenizer
+from dataclasses import dataclass
+from typing import List, Tuple
+
+import evaluate
+from evaluate import logging
+from rageval.metrics import Metric, add_attribute
+
+
+_CITATION = """\
```
Comment (QianHaosheng) [inline rageval/metrics/_answer_precision.py:14]: This citation doesn't seem to be correct.

File: rageval/metrics/_answer_precision.py [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+        so the perplexity can include the probability of the first word. Defaults to True.
+    device (str): device to run on, defaults to 'cuda' when available
+
+Returns:
+    perplexity: dictionary containing the perplexity scores for the texts
+        in the input list, as well as the mean perplexity. If one of the input texts is
+        longer than the max input length of the model, then it is truncated to the
+        max length for the perplexity computation.
+
+Examples:
```
Comment (QianHaosheng) [inline rageval/metrics/_answer_precision.py:51]: This makes it difficult to pass CI tests in an environment without a GPU.

File: rageval/metrics/_answer_precision.py [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+import torch
+from torch.nn import CrossEntropyLoss
+from transformers import AutoModelForCausalLM, AutoTokenizer
+from dataclasses import dataclass
+from typing import List, Tuple
+
+import evaluate
+from evaluate import logging
+from rageval.metrics import Metric, add_attribute
+
```
Comment (QianHaosheng) [inline rageval/metrics/_answer_precision.py:12]: `import/from` is not standardized here

File: rageval/metrics/_answer_precision.py [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+                }
+            ),
+            reference_urls=["https://huggingface.co/docs/transformers/perplexity"]
+        )
+
+    def _compute(
+        self,
+        predictions: List[str],
+        pipeline,
+    ) -> Tuple[float, List[float]]:
```
Comment (bugtig6351) [inline rageval/metrics/_answer_precision.py:125]: Where is the entry point of the `_compute()` method?

**Review by faneshion — APPROVED** [review]
